pkg/auditor/auditor_test.go

```diff
@@ -52,60 +52,73 @@ func (suite *AuditorTestSuite) TearDownTest() {
 func (suite *AuditorTestSuite) TestAuditorUpdatesRegistry() {
 	suite.a.registry = make(map[string]*RegistryEntry)
 	suite.Equal(0, len(suite.a.registry))
-	suite.a.updateRegistry(suite.source.Path, 42)
+	suite.a.updateRegistry(suite.source.Path, 42, nil)
 	suite.Equal(1, len(suite.a.registry))
 	suite.Equal(int64(42), suite.a.registry[suite.source.Path].Offset)
-	suite.a.updateRegistry(suite.source.Path, 43)
-	suite.Equal(int64(43), suite.a.registry[suite.source.Path].Offset)
+	suite.Equal(time.Time{}, suite.a.registry[suite.source.Path].Timestamp)
+	suite.a.updateRegistry(suite.source.Path, 43, nil)
+	ts := time.Now()
+	suite.a.updateRegistry("containerid", 0, &ts)
+	suite.Equal(ts, suite.a.registry["containerid"].Timestamp)
 }
 
 func (suite *AuditorTestSuite) TestAuditorFlushesAndRecoversRegistry() {
 	suite.a.registry = make(map[string]*RegistryEntry)
 	suite.a.registry[suite.source.Path] = &RegistryEntry{
-		Path:      suite.source.Path,
-		Timestamp: time.Date(2006, time.January, 12, 1, 1, 1, 1, time.Local),
-		Offset:    42,
+		LastUpdated: time.Date(2006, time.January, 12, 1, 1, 1, 1, time.Local),
+		Offset:      42,
 	}
 	suite.a.flushRegistry(suite.a.registry, suite.testPath)
 	r, err := ioutil.ReadFile(suite.testPath)
 	suite.Nil(err)
-	suite.Equal("{\"Version\":0,\"Registry\":{\"testpath\":{\"Path\":\"testpath\",\"Timestamp\":\"2006-01-12T01:01:01.000000001Z\",\"Offset\":42}}}", string(r))
+	suite.Equal("{\"Version\":0,\"Registry\":{\"testpath\":{\"Offset\":42,\"Timestamp\":\"0001-01-01T00:00:00Z\",\"LastUpdated\":\"2006-01-12T01:01:01.000000001Z\"}}}", string(r))
```

Comment:
we update the auditor payload. While agent doesn't crash, I need to do more testing on the exact behavior

Reply:
See this part of PR description:

Note on these two changes: they aren't backward compatible per say, but nothing will break. When agent restarts with new code, we cleanup the registry, so we lose any offset and we start tailing from end. This means that we may lose a few log lines, as we won't restart exactly where we stopped. I think we're fine with as we're in beta.
An easy alternative is to schedule the cleanup after 1 minute, which would reduce problems (but not eliminate) at the cost of some small code readability decrease.
An alternative to completely be backwards compatible is possible, but would be a big decrease of readability.